src/Stratis.Bitcoin.Features.Consensus/PowConsensusValidator.cs

```diff
@@ -331,9 +376,20 @@ protected virtual void CheckMaturity(UnspentOutputs coins, int nSpendHeight)
             this.logger.LogTrace("(-)");
         }
 
-        public virtual void CheckInputs(Transaction tx, UnspentOutputSet inputs, int nSpendHeight)
+        /// <summary>
+        /// Checks that transaction's inputs are valid.
+        /// </summary>
+        /// <param name="tx">The transaction.</param>
+        /// <param name="inputs">Inputs for the transaction.</param>
```

Comment:
this is imho incorrect
from the code, this seems to be UTXO set 
you have UTXO set that defines who has which coins and if a transaction has inputs that are not in UTXO set, tx is invalid